crates/uv-pep508/src/marker/tree.rs

```diff
@@ -1002,11 +1003,19 @@ impl MarkerTree {
     ///
     /// ASSUMPTION: There is one `extra = "..."`, and it's either the only marker or part of the
     /// main conjunction.
-    pub fn top_level_extra_name(self) -> Option<ExtraName> {
+    pub fn top_level_extra_name(self) -> Option<Cow<'static, ExtraName>> {
```

Comment:
If we didn't go through the DNF for the general top level marker case this could be a `&'static ExtraName`